java/ql/lib/semmle/code/java/internal/OverlayXml.qll

```diff
@@ -0,0 +1,45 @@
+overlay[local]
+module;
+
+/**
+ * A local predicate that always holds for the overlay variant and never holds for the base variant.
+ * This is used to define local predicates that behave differently for the base and overlay variant.
+ */
+private predicate isOverlay() { databaseMetadata("isOverlay", "true") }
+
+private @file getXmlFile(@xmllocatable locatable) {
```

Comment:
Do all languages extract `@file` using named trap ids? Because if not, then this identification change from string to `@file` won't work. I'm guessing that they do, but this is a subtle change, so I figured it would be best to check.